File: javascript/lasagna-master/lasagna-master.js
/// <reference path="./global.d.ts" />
// @ts-check

/**
 * Implement the functions needed to solve the exercise here.
 * Do not forget to export them so they are available for the
 * tests. Here an example of the syntax as reminder:
 *
 * export function yourFunction(...) {
 *   ...
 * }
 */

export function cookingStatus(time) {
  switch(time) {
    case 0:
      return 'Lasagna is done.';
    case undefined:
      return 'You forgot to set the timer.'
    default:
      return 'Not done, please wait.'
  }
}

export function preparationTime(layers, avg = 2) {
  return layers.length * avg;
}

export function quantities(layers) {
  let noodles = 0;
  let sauce = 0;
  layers.forEach((layer) => {
    if (layer === "noodles") {
      noodles += 1;
    }
    if (layer === "sauce") {
      sauce += 1;
    }
  })

  return { noodles: noodles * 50, sauce: sauce * 0.2 }
}

export function addSecretIngredient(friendList, myList) {
  myList.push(friendList[friendList.length - 1]);
}

export function scaleRecipe(recipe, portions) {
  const newRecipe = {};

  Object.keys(recipe).forEach((r) => {
    newRecipe[r] = (recipe[r] / 2) * portions;
  })

  return newRecipe;
}

File: go/lasagna-master/lasagna_master.go
package lasagna


const DefaultPreparationTime = 2

// TODO: define the 'PreparationTime()' function
func PreparationTime(layers []string, count int) int {
  if count == 0 {
    count = DefaultPreparationTime
  }

  return len(layers) * count
}
// TODO: define the 'Quantities()' function
func Quantities(ingredients []string) (noodles int, sauce float64){
  for _, value := range ingredients {
    if value == "sauce" {
      sauce += 0.2
    }
    if value == "noodles" {
      noodles += 50
    }
  }
  return
}

// TODO: define the 'AddSecretIngredient()' function
func AddSecretIngredient(friendList []string, myList []string) {
  myList[len(myList) - 1] = friendList[len(friendList) - 1]
}

// TODO: define the 'ScaleRecipe()' function
func ScaleRecipe(quantities []float64, mult int) []float64 {
  result := []float64{}
  for i := range quantities {
    result = append(result, (quantities[i] / 2.0 ) * float64(mult))
  }

  return result
}

// Your first steps could be to read through the tasks, and create
// these functions with their correct parameter lists and return types.
// The function body only needs to contain `panic("")`.
// 
// This will make the tests compile, but they will fail.
// You can then implement the function logic one by one and see
// an increasing number of tests passing as you implement more 
// functionality.
